src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -34,15 +35,21 @@ public OrderController(ShoppingCartService shoppingCartService,
     }
 
     @PostMapping("/complete")
-    public OrderResponseDto completeOrder(@RequestParam Long userId) {
-        ShoppingCart cart = shoppingCartService.getByUser(userService.get(userId));
+    public OrderResponseDto completeOrder(@RequestParam Authentication authentication) {
+        ShoppingCart cart = shoppingCartService.getByUser(userService
+                .findByEmail(authentication.getName())
+                .orElseThrow(() -> new RuntimeException("Can't authenticate a user: "
+                        + authentication.getName())));
         return orderResponseDtoMapper.mapToDto(orderService.completeOrder(cart));
     }
 
     @GetMapping
-    public List<OrderResponseDto> getOrderHistory(@RequestParam Long userId) {
-        return orderService.getOrdersHistory(userService.get(userId))
-                .stream()
+    public List<OrderResponseDto> getOrderHistory(@RequestParam Authentication authentication) {
```

Comment:
```suggestion
    public List<OrderResponseDto> getOrderHistory(Authentication authentication) {
```
fix in other methods accordingly